expression/builtin_time.go

```diff
@@ -5866,18 +5866,21 @@ func (c *makeTimeFunctionClass) getFunction(ctx sessionctx.Context, args []Expre
 	}
 	tp, flen, decimal := args[2].GetType().EvalType(), 10, 0
 	switch tp {
-	case types.ETInt:
-	case types.ETReal, types.ETDecimal:
-		decimal = args[2].GetType().Decimal
-		if decimal > 6 || decimal == types.UnspecifiedLength {
-			decimal = 6
+	case types.ETString:
```

Comment:
```
  bool resolve_type(THD *) override {
    set_data_type_time(MY_MIN(args[2]->decimals, DATETIME_MAX_DECIMALS));
    return false;
  }
```

I think this can work, we don't need getExpressionFsp().

Reply:
But be careful with UnspecifiedLength, which is 31 in MySQL but -1 in TiDB

Reply:
thanks for you reply,  I will continue after your works finished

Reply:
If the type is ETSTRING, then the decimal should be -1?

Reply:
Can we set the decimal be `UnspecifiedLength`, if the type is `ETSTRING` ?